swarm/api/act.go

```diff
@@ -261,18 +267,29 @@ func (a *API) doDecrypt(ctx context.Context, credentials string, pk *ecdsa.Priva
 			m.Access = nil
 			return nil
 		case "act":
-			publisherBytes, err := hex.DecodeString(m.Access.Publisher)
-			if err != nil {
-				return ErrDecrypt
-			}
-			publisher, err := crypto.DecompressPubkey(publisherBytes)
-			if err != nil {
-				return ErrDecrypt
-			}
+			var (
```

Comment:
not sure what behaviour this code has. 
What we need is to try generate the session key using pk for the current bzzaccount. 
If this fails, we fall back to password